src/lib/operators/join_index.cpp

```diff
@@ -55,8 +57,21 @@ std::shared_ptr<const Table> JoinIndex::_on_execute() {
                   !_secondary_predicates.empty()),
          "JoinHash doesn't support these parameters");
 
-  _right_matches.resize(input_table_right()->chunk_count());
-  _left_matches.resize(input_table_left()->chunk_count());
+  std::shared_ptr<const Table> left_input_table;
+  std::shared_ptr<const Table> right_input_table;
+  auto primary_predicate = _primary_predicate;
+
+  if (_index_side == JoinInputSide::Left) {
+    left_input_table = input_table_right();
```

Comment:
Same as with the hash join (#1557) : Once we swap the tables, we should stop talking about "left" and "right" sides. Maybe "indexed" and "probed" side?